### ISSUE TITLE
Unexpected key when load pretrained model, attr_prob seems random.

### ISSUE THREAD

**Issue by gyang274** [open-mmlab/mmfashion#12]
Thank you for your great work!

I followed `demo/test_predictor.py` to load the model using:
```
model = build_predictor(cfg.model)
checkpoint = load_checkpoint(model, </path/to/resnet50/pretrained/model/latest.pth>, map_location='cpu')
```
And got:
```
The model and loaded state dict do not match exactly

unexpected key in source state_dict: roi_pool.linear.0.weight, roi_pool.linear.0.bias, concat.fc_fusion.weight, concat.fc_fusion.bias
```

The prediction on `demo/attr_pred_demo1.jpg` seems random since all probs are 0.4-0.6:
```
attr_prob = model(img_tensor, attr=None, landmark=None, return_loss=False)

attr_prob

tensor([[0.5356, 0.5116, 0.5132, 0.4532, 0.5143, 0.4566, 0.4571, 0.4589, 0.4532,
         0.5084, 0.4872, 0.4767, 0.4837, 0.5268, 0.3955, 0.4436, 0.5487, 0.4131,
         0.3939, 0.4990, 0.5329, 0.5224, 0.4197, 0.4822, 0.4122, 0.4306, 0.5518,
         0.4582, 0.5110, 0.4674, 0.4495, 0.4863, 0.4816, 0.5129, 0.5425, ...,
         0.4632]], device='cuda:0', grad_fn=<SigmoidBackward>)
```

The cfg.model is the default one:
```
cfg.model                                                                                                                                                  
Out[42]: 
{'type': 'GlobalPredictor',
 'backbone': {'type': 'ResNet'},
 'global_pool': {'type': 'GlobalPooling',
  'inplanes': (7, 7),
  'pool_plane': (2, 2),
  'inter_channels': [2048, 4096],
  'outchannels': 4096},
 'attr_predictor': {'type': 'AttrPredictor',
  'inchannels': 4096,
  'outchannels': 1000},
 'loss_attr': {'type': 'BCEWithLogitsLoss',
  'ratio': 1,
  'weight': None,
  'size_average': None,
  'reduce': None,
  'reduction': 'mean'},
 'pretrained': 'checkpoint/resnet50.pth'}
```

Any suggestion? What is the expected output for `demo/attr_pred_demo1.jpg`?

Thank you.

**Comment by gyang274** [comment]
Mis-used landmark pooling checkpoint for global pooling.

**Comment by gyang274** [comment]
Have to reopen this, because the attr_prob still seems random even after matched the model and checkpoint.

Model Config: `configs/attribute_predict/global_predictor_resnet_attr.py`

Checkpoint: `checkpoint/Predict/resnet/global/model_best.pth`

Input: `demo/attr_pred_demo1.jpg`

attr_prob
``` 
tensor([[0.0104, 0.0152, 0.0004, 0.0002, 0.0003, 0.0013, 0.0007, 0.0006, 0.0005,
         0.0003, 0.0005, 0.0071, 0.0006, 0.0004, 0.0053, 0.0046, 0.0004, 0.0014,
         0.0036, 0.0022, 0.0031, 0.0008, 0.0005, 0.0012, 0.0022, 0.0033, 0.0006,
         0.0024, 0.0008, 0.0006, 0.0040, 0.0003, 0.0005, 0.0003, 0.0004, 0.0003,
         0.0031, 0.0007, 0.0030, 0.0020, 0.0010, 0.0008, 0.0034, 0.0014, 0.0048,
         0.0042, 0.0004, 0.0024, 0.0047, 0.0008, 0.0073, 0.0006, 0.0003, ....
         0.0016]], device='cuda:0', grad_fn=<SigmoidBackward>)
```

Top5 Prediction:
```
Label: print, Probs: 0.0011296379379928112.
Label: knit, Probs: 0.0010909560369327664.
Label: lace, Probs: 0.0010880372719839215.
Label: sleeve, Probs: 0.0010854118736460805.
Label: shirt, Probs: 0.0010577059583738446.
```

**Comment by veralauee** [comment]
> Have to reopen this, because the attr_prob still seems random even after matched the model and checkpoint.
> 
> Model Config: `configs/attribute_predict/global_predictor_resnet_attr.py`
> 
> Checkpoint: `checkpoint/Predict/resnet/global/model_best.pth`
> 
> Input: `demo/attr_pred_demo1.jpg`
> 
> attr_prob
> 
> ```
> tensor([[0.0104, 0.0152, 0.0004, 0.0002, 0.0003, 0.0013, 0.0007, 0.0006, 0.0005,
>          0.0003, 0.0005, 0.0071, 0.0006, 0.0004, 0.0053, 0.0046, 0.0004, 0.0014,
>          0.0036, 0.0022, 0.0031, 0.0008, 0.0005, 0.0012, 0.0022, 0.0033, 0.0006,
>          0.0024, 0.0008, 0.0006, 0.0040, 0.0003, 0.0005, 0.0003, 0.0004, 0.0003,
>          0.0031, 0.0007, 0.0030, 0.0020, 0.0010, 0.0008, 0.0034, 0.0014, 0.0048,
>          0.0042, 0.0004, 0.0024, 0.0047, 0.0008, 0.0073, 0.0006, 0.0003, ....
>          0.0016]], device='cuda:0', grad_fn=<SigmoidBackward>)
> ```
> 
> Top5 Prediction:
> 
> ```
> Label: print, Probs: 0.0011296379379928112.
> Label: knit, Probs: 0.0010909560369327664.
> Label: lace, Probs: 0.0010880372719839215.
> Label: sleeve, Probs: 0.0010854118736460805.
> Label: shirt, Probs: 0.0010577059583738446.
> ```

The "attr_prob" seemingly does not match "top5 Prediction". The highest prob is not 0.0011296379379928112.

**Comment by gyang274** [comment]
@veralauee I applied a softmax on the attr_prob. It seems attr_prob are not probabilities, since sum(attr_prob) not equal to 1.

```
# predict probabilities for each attribute
attr_prob = model(img_tensor, attr=None, landmark=None, return_loss=False)

probs = torch.nn.functional.softmax(attr_prob[0], dim=0)
  
top5 = torch.topk(probs, 5)
  
attr_cloth_file = open(os.path.join('../..', cfg.data.test.attr_cloth_file)).readlines()

attr_idx2name = {}
for i, line in enumerate(attr_cloth_file[2:]):
  attr_idx2name[i] = line.strip('\n').split()[0]

for (i, v) in zip(top5.indices, top5.values):
  print(f"Label: {attr_idx2name[i.item()]}, Probs: {v.item()}.")
```

**Comment by veralauee** [comment]
In attribute prediction task, each image(garment) has multiple attributes, e.g, T-shirt and print. 
Softmax is not applicable for multi-label prediction task.

**Comment by veralauee** [comment]
If an image(garment) contains a speific attribute, the prob of that attribute should be 1. Thus, in multi-label prediction, the sum of prob will be more than 1.

**Comment by liuziwei7** [comment]
Will close this issue for now. Please feel free to re-open it in case of any further questions.
